sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/SyntaxHelper.cs

```diff
@@ -202,5 +202,18 @@ public static int GetLineNumber(this SyntaxToken token)
         {
             return token.GetLocation().GetLineSpan().StartLinePosition.Line;
         }
+
+        public static bool IsCatchException(this CatchClauseSyntax catchClause)

```

Comment:
Split this into `IsEmptyCatch` and `IsCatchingRootException`/`IsCatchingAllExceptions`